test/jasmine/tests/histogram_test.js

```diff
@@ -186,6 +186,343 @@ describe('Test histogram', function() {
+            }
+        }
+
+        it('should group traces w/ same axes and w/ same orientation', function() {
+            var barModes = ['group', 'stack'];
+
+            barModes.forEach(function(mode) {
+                gd = {
+                    data: [
+                        {type: 'histogram', y: [1]},
+                        {type: 'histogram', y: [2]},
+
+                        {type: 'histogram', y: [1], xaxis: 'x2'},
+                        {type: 'histogram', y: [3], xaxis: 'x2'},
+
+                        {type: 'histogram', y: [3]},
+                        {type: 'histogram', y: [2], xaxis: 'x2'},
+
+                        {type: 'histogram', x: [1]},
+                        {uid: 'solo', type: 'histogram', x: [2], yaxis: 'y2'},
+                        {type: 'histogram', x: [2]}
+                    ],
+                    layout: {barmode: mode}
+                };
+                supplyAllDefaults(gd);
+                _assert('under barmode:' + mode, [
+                    ['xyy', [0, 1, 4]],
+                    ['x2yy', [2, 3, 5]],
+                    ['xyx', [6, 8]],
+                    ['solo__x', [7]]
+                ]);
+            });
+        });
+
+        it('should group traces on matching axes and w/ same orientation', function() {
+            var barModes = ['group', 'stack'];
+
+            barModes.forEach(function(mode) {
+                gd = {
+                    data: [
+                        {type: 'histogram', y: [1]},
+                        {type: 'histogram', y: [2], xaxis: 'x2'},
+                        {type: 'histogram', x: [1], yaxis: 'y2'},
+                        {type: 'histogram', x: [2], yaxis: 'y2'},
+                    ],
+                    layout: {
+                        barmode: mode,
+                        xaxis2: {matches: 'x'},
+                        yaxis2: {matches: 'x'}
+                    }
+                };
+                supplyAllDefaults(gd);
+                _assert('under barmode:' + mode, [
+                    ['g0yy', [0, 1]],
+                    ['g0g0x', [2, 3]]
+                ]);
+            });
+        });
+
+        it('should not group traces by default under barmode:overlay ', function() {
+            gd = {
+                data: [
+                    {uid: 'a', type: 'histogram', y: [1]},
+                    {uid: 'b', type: 'histogram', y: [2]},
+
+                    {uid: 'c', type: 'histogram', y: [1], xaxis: 'x2'},
+                    {uid: 'd', type: 'histogram', y: [3], xaxis: 'x2'},
+
+                    {uid: 'e', type: 'histogram', y: [3]},
+                    {uid: 'f', type: 'histogram', y: [2], xaxis: 'x2'},
+
+                    {uid: 'g', type: 'histogram', x: [1]},
+                    {uid: 'h', type: 'histogram', x: [2], yaxis: 'y2'},
+                    {uid: 'i', type: 'histogram', x: [2]}
+                ],
+                layout: {barmode: 'overlay'}
+            };
```

Comment:
Interestingly, this mock does not render in this PR and in `master`. :thinking: 

![Screenshot_2019-05-17_17-13-22](https://user-images.githubusercontent.com/301546/57956487-1bb23680-78c7-11e9-84d2-4489df285061.png)